### PR TITLE
Update to 1.19

### PR DESCRIPTION
**Pull request by tambry**
* Correctly specify snowy coniferous forest as a taiga.
* Biome categories converted to Fabric tags.
* Add a fir boat with chest.
* Fixed door rendering to match vanilla.

~~Can't target this against a 1.19 branch as one doesn't exist. Let me know if you'd like me to create another PR if you create one.~~ Feel free to also push this yourself, but please keep me as the committer.

### CODE REVIEW

File: common/src/main/java/com/terraformersmc/traverse/item/TraverseBoatTypes.java [hunk 1]
```diff
@@ -4,16 +4,24 @@
 import com.terraformersmc.terraform.boat.api.TerraformBoatTypeRegistry;
 import com.terraformersmc.terraform.boat.api.item.TerraformBoatItemHelper;
 import com.terraformersmc.traverse.Traverse;
+import java.util.function.Supplier;
 import net.minecraft.util.Identifier;
 import net.minecraft.util.registry.Registry;
 
 public class TraverseBoatTypes {
 	public static TerraformBoatType fir;
+	public static TerraformBoatType fir_chest;
```
Comment (haykam821) [inline common/src/main/java/com/terraformersmc/traverse/item/TraverseBoatTypes.java]: The same boat type should be used for both boats and chest boats.

File: client/src/main/resources/assets/traverse/lang/en_us.json [hunk 2]
```diff
@@ -45,5 +45,7 @@
   "block.traverse.fir_door": "Fir Door",
   "block.traverse.fir_trapdoor": "Fir Trapdoor",
   "item.traverse.fir_boat": "Fir Boat",
-  "entity.traverse.fir_boat": "Fir Boat"
+  "item.traverse.fir_chest_boat": "Fir Boat with Chest",
+  "entity.traverse.fir_boat": "Fir Boat",
+  "entity.traverse.fir_chest_boat": "Fir Boat with Chest"
```
Comment (haykam821) [inline client/src/main/resources/assets/traverse/lang/en_us.json]: These translations are no longer necessary as Terraform handles the boat entity types.

**Comment by tambry** [comment]
Comments adressed, thanks for the quick review. 🙂

File: common/src/main/java/com/terraformersmc/traverse/item/TraverseBoatTypes.java [hunk 3]
```diff
@@ -11,7 +11,8 @@ public class TraverseBoatTypes {
 	public static TerraformBoatType fir;
 
 	public static void register() {
-		var item = TerraformBoatItemHelper.registerBoatItem(new Identifier(Traverse.MOD_ID, "fir_boat"), () -> fir);
+		var item = TerraformBoatItemHelper.registerBoatItem(new Identifier(Traverse.MOD_ID, "fir_boat"), () -> fir, false);
+		TerraformBoatItemHelper.registerBoatItem(new Identifier(Traverse.MOD_ID, "fir_chest_boat"), () -> fir, true);
 
 		fir = new TerraformBoatType.Builder().item(item).build();
```
Comment (haykam821) [inline common/src/main/java/com/terraformersmc/traverse/item/TraverseBoatTypes.java]:
```suggestion
		var chestItem = TerraformBoatItemHelper.registerBoatItem(new Identifier(Traverse.MOD_ID, "fir_chest_boat"), () -> fir, true);

		fir = new TerraformBoatType.Builder()
				.item(item)
				.chestItem(chestItem)
				.build();
```

**Review by haykam821 — COMMENTED** [review]
Just one more change should be needed for chest boats to work, which is defining their item (which is used as the pick stack).

**Comment by NeusFear** [comment]
Haven't had time to look at this fully, but I did make a 1.19 branch for you to set base on

File: common/src/main/java/com/terraformersmc/traverse/item/TraverseBoatTypes.java (hunk 3 above)
Comment (tambry) [reply common/src/main/java/com/terraformersmc/traverse/item/TraverseBoatTypes.java]: Good catch, thanks!

**Comment by gniftygnome** [comment]
Could you PR the biome tag changes against 1.18.2?  They almost apply straight across but I can't seem to get IDEA to pick just that commit from this PR while keeping you as committer.

**Comment by tambry** [comment]
> Could you PR the biome tag changes against 1.18.2? They almost apply straight across but I can't seem to get IDEA to pick just that commit from this PR while keeping you as committer.

Done, see #88. Hopefully that's what you meant.

**Comment by gniftygnome** [comment]
Yes, I hadn't gotten to that yet but if it's going into 1.19 it might as well be in 1.18.2 as well.  Today's work on 1.18.2 Traverse hopefully gets us to a stable release.  Then it'll be easier for me to look at the 1.19 version.

**Comment by gniftygnome** [comment]
Thanks!

**Pull request merged**